### PR TITLE
Tokens should only be used in the same site as the oauth client object

### PR DESCRIPTION
**Pull request by felipemontoya**
With this PR the token client is compared to the host. If they don't match, then it is the same as if no authorization had happened.

This is critical for multi-tenant access.

SessionAuthorization is no longer possible unless the user is_staff = true

### CODE REVIEW

File: eox_core/api/v1/permissions.py [hunk 1]
```diff
@@ -32,10 +32,22 @@ def load_permissions():
...
+        try:
+            allowed_for_site = request.user.is_staff or request.get_host() in request.auth.client.url
+        except Exception:  # pylint: disable=broad-except
+            allowed_for_site = False
+
+        if not allowed_for_site:
+            # If we get here either someone is using a token created on one site in a different site
+            # Or there was a missconfiguration of the oauth client.
+            # To prevent leaking important information we return the most basic message
+            raise exceptions.NotAuthenticated(detail="Invalid token")
```
Comment (Squirrel18) [inline eox_core/api/v1/permissions.py]: Maybe, you can raise this exception on line 45, since 'allowed_for_site' is only used to raise this exception. @felipemontoya

File: eox_core/api/v1/permissions.py [hunk 2]
```diff
@@ -32,10 +32,22 @@ def load_permissions():
 class EoxCoreAPIPermission(permissions.BasePermission):
     """
     Defines a custom permissions to access eox-core API
+    This permissions makes sure that a token is created with the client credentias of the same site is being used on
```
Comment (Squirrel18) [inline eox_core/api/v1/permissions.py]: Please, end the line comments with a period. @felipemontoya

File: eox_core/api/v1/permissions.py (hunk 1 above)
Comment (felipemontoya) [reply eox_core/api/v1/permissions.py]: I had that first, but when I added the user.is_staff check the line's logic became hard to follow

File: eox_core/api/v1/permissions.py [hunk 3]
```diff
@@ -32,10 +32,22 @@ def load_permissions():
 class EoxCoreAPIPermission(permissions.BasePermission):
     """
     Defines a custom permissions to access eox-core API
+    This permissions makes sure that a token is created with the client credentias of the same site is being used on.
```
Comment (anmrdz) [inline eox_core/api/v1/permissions.py]:
```suggestion
    These permissions make sure that a token is created with the client credentials of the same site is being used on.
```

File: eox_core/api/v1/permissions.py [hunk 4]
```diff
@@ -32,10 +32,22 @@ def load_permissions():
...
     """
 
     def has_permission(self, request, view):
         """
         To grant accees, checks if the request user either can call eox-core API or if it's an admin user.
```
Comment (anmrdz) [inline eox_core/api/v1/permissions.py]:
```suggestion
        To grant access, checks if the requesting user either can call eox-core API or if it's an admin user.
```
